### PR TITLE
feat: remove dists site-packages's when python interpreter changes

### PR DESCRIPTION
**Pull request by tdejager**
Remove distribution from site-packages dir when interpreter changes.

### CODE REVIEW

File: src/install_pypi.rs [hunk 1]
```diff
@@ -271,6 +272,45 @@ fn whats_the_plan<'a>(
...
+            .await
+            .expect("unistallation of old site-packages failed");
+        println!(
+            "Uninstalled {} ({} file{}, {} director{})",
+            dist_info.name(),
+            summary.file_count,
+            if summary.file_count == 1 { "" } else { "s" },
+            summary.dir_count,
+            if summary.dir_count == 1 { "y" } else { "ies" },
+        );
```
Comment (baszalmstra) [inline src/install_pypi.rs:308]:
```suggestion
```

File: src/install_pypi.rs [hunk 2]
```diff
@@ -289,26 +321,22 @@ pub async fn update_python_distributions(
         return Ok(());
     };
 
-    let python_location = prefix.root().join(&python_info.path);
+    let platform = platform_host::Platform::current().expect("unsupported platform");
 
-    // Determine where packages would have been installed
-    let _python_version = (python_info.short_version.1 as u32, 0);
+    dbg!(&status);
```
Comment (baszalmstra) [inline src/install_pypi.rs:326]:
```suggestion
```

**Review by baszalmstra — APPROVED** [review]

**Comment by ruben-arts** [comment]
This pr triggered me to checkout the folders some more. And if found this situation:
```
cd examples/pypi
pixi add python==3.10
➜ ls .pixi/envs/default/lib/ | rg python
libpython3.10.so
libpython3.10.so.1.0
libpython3.so
python3.1
python3.10
python3.11
```
Which seems weird to me. Does this have anything to do with this PR? Otherwise I'll put it in an issue.

**Comment by tdejager** [comment]
> This pr triggered me to checkout the folders some more. And if found this situation:
> 
> ```
> cd examples/pypi
> pixi add python==3.10
> ➜ ls .pixi/envs/default/lib/ | rg python
> libpython3.10.so
> libpython3.10.so.1.0
> libpython3.so
> python3.1
> python3.10
> python3.11
> ```
> 
> Which seems weird to me. Does this have anything to do with this PR? Otherwise I'll put it in an issue.

No it should not, all those are conda executables.

**Pull request merged**